### PR TITLE
add the main_ground_data pointer

### PR DESCRIPTION
**Pull request by marius851000**
That's a pretty simple change, to test a bit the CI (but entered information are valid, at least AFAIK)

### CODE REVIEW

**Comment by UsernameFodder** [comment]
The CI will be a bit annoying on this first PR since you're a first-time contributor and I'll have to approve runs (apparently GitHub does this to prevent cryptomining abuse), but in future PRs it should be easier.

File: headers/types/ground_mode/ground_mode.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Types used during the ground part (overlay 11)
+
+#ifndef HEADERS_TYPES_GROUND_MODE_H_
+#define HEADERS_TYPES_GROUND_MODE_H_
+
+#include "partner_following.h";
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:6]: There's no such file. Delete this line?

File: headers/types/ground_mode/ground_mode.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// A global structure holding various pointer to important structure
+struct main_ground_data {
+    undefined* script; //0x0: pointer to script structure
+    undefined* partner_follow_data; //0x4: pointer to the data related to the partner following the player
+    undefined[]* actors; //0x8: pointer to the actors
+    undefined[]* objects; //0x12: pointer to the objects
+    undefined[]* performers; //0x16: pointer to the performers
+    undefined[]* events; //0x20: pointer to the events
+}
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:16]: Semicolon here

File: headers/types/ground_mode/ground_mode.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#ifndef HEADERS_TYPES_GROUND_MODE_H_
+#define HEADERS_TYPES_GROUND_MODE_H_
+
+#include "partner_following.h";
+
+// A global structure holding various pointer to important structure
+struct main_ground_data {
+    undefined* script; //0x0: pointer to script structure
+    undefined* partner_follow_data; //0x4: pointer to the data related to the partner following the player
+    undefined[]* actors; //0x8: pointer to the actors
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:12]: I don't think there's much point in making this an array pointer `[]*`. Let's just make it a normal pointer.

File: headers/types/ground_mode/ground_mode.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Types used during the ground part (overlay 11)
+
+#ifndef HEADERS_TYPES_GROUND_MODE_H_
+#define HEADERS_TYPES_GROUND_MODE_H_
+
+#include "partner_following.h";
+
+// A global structure holding various pointer to important structure
+struct main_ground_data {
+    undefined* script; //0x0: pointer to script structure
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:10]: Some [formatting issues](https://github.com/UsernameFodder/pmdsky-debug/runs/5336315279?check_suite_focus=true) with the comments. Need a space after the `//`. You can also [download the formatted file](https://github.com/UsernameFodder/pmdsky-debug/actions/runs/1898774927#artifacts) directly.

File: headers/types/ground_mode/ground_mode.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Types used during the ground part (overlay 11)
+
+#ifndef HEADERS_TYPES_GROUND_MODE_H_
+#define HEADERS_TYPES_GROUND_MODE_H_
+
+#include "partner_following.h";
+
+// A global structure holding various pointer to important structure
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:8]: Can we elaborate a bit more? How about "A global structure holding various pointers to important structures for ground mode"?

File: symbols/overlay11.yml [hunk 6]
```diff
@@ -114,5 +114,9 @@ overlay11:
...
         EU: 0x2325834
       length:
-        NA: 0x14
-        EU: 0x14
+        NA: 0x18
+        EU: 0x18
+      description: |-
+        Host pointers to multiple structure used for performing an overworld scene
+        
+        type: main_ground_data
```
Comment (UsernameFodder) [inline symbols/overlay11.yml:122]: `struct main_ground_data`

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

**Comment by marius851000** [comment]
Well, finally applied those change (after having the motivation following the discussion in the new asm channel).

File: headers/types/types.h [hunk 7]
```diff
@@ -4,5 +4,4 @@
 #include "common/common.h"
 #include "dungeon_mode/dungeon_mode.h"
 #include "ground_mode/ground_mode.h"
-
```
Comment (UsernameFodder) [inline headers/types/types.h:7]: Can we leave this file as is? There's no need to change anything here.

File: headers/types/ground_mode/ground_mode.h [hunk 8]
```diff
@@ -99,4 +99,15 @@ struct script_object {
 };
 ASSERT_SIZE(struct script_object, 12);
 
+// A global structure holding various pointer to important structure for ground mode
+struct main_ground_data {
+    undefined* script; // 0x0: pointer to script structure
```
Comment (UsernameFodder) [inline headers/types/ground_mode/ground_mode.h:104]: The formatter likes comments on struct fields to be aligned; you can just download [the formatted file (under "Artifacts")](https://github.com/UsernameFodder/pmdsky-debug/actions/runs/2237612286#artifacts) and apply it to this PR.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
I almost forgot, make sure to add yourself to the top-level `CREDITS.txt` file if you want! 😄

**Comment by marius851000** [comment]
For the credit, should I put myself under ``SBDWolf`` or after the other generic credits ?

**Comment by marius851000** [comment]
Humm...The formatter did some strange stuff, but hopefully, it should work now.

**Comment by UsernameFodder** [comment]
> For the credit, should I put myself under `SBDWolf` or after the other generic credits ?

Yep, that's good.

> Humm...The formatter did some strange stuff, but hopefully, it should work now.

Yeah there's a soft column limit, so the formatter opted to wrap the line. Usually I move the comment to the line above in cases like that, but it doesn't matter too much.

**Review by UsernameFodder — APPROVED** [review]
LGTM, thanks!

**Pull request merged**